tests/test_sort_product_functionality.py

```diff
@@ -22,12 +30,14 @@ def test_sort_product_a_z_functionality(self, login_to_saucedemo):
             assert product_names == sorted(product_names), "Products are not sorted alphabetically (A to Z)"
 
     def test_sort_product_z_a_functionality(self, login_to_saucedemo):
-        WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.CLASS_NAME, "inventory_list")))
+        WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.XPATH, '//div[@class="inventory_list"]')))
 
         sort_option = driver.find_element(By.XPATH, '//span[@class="select_container"]')
+        assert sort_option.is_displayed(), "The button is not visible"
         sort_option.click()
 
         sort_option_z_a = driver.find_element(By.XPATH, '//option[text()="Name (Z to A)"]')
+        assert sort_option_z_a.is_displayed(), "The button is not visible"
```

Comment:
@Sv-24 what are the difference between this tests and previous one? how do u know what sorting is displayed from z to a, or from a to z?

Reply:
@irynalavrinok-pango  The main difference is that each method selects a different sorting option: one for sorting from A to Z ("Name (A to Z)") and the other for sorting from Z to A ("Name (Z to A)"). But I've added some changes. Before clicking on the sorting (A to Z), I check the sorting status, and after clicking sorting, I have retrieved product names, and then I check the sorting status again. Also, I've added "reverse=True" to the method( Z to A) that ensures that sorting is performed in reverse order.

Reply:
@Sv-24 got it. thanks.